### DISCUSSION TITLE
Smoothing of trajectories

### DISCUSSION THREAD

**Discussion by chraibi** [PedestrianDynamics/PedPy#363] [Ideas]
In many papers, we see that there is a need to pre-process the trajectories by applying smoothing filters like for example Savitzky-Golay filter.

Maybe we could add a class of filters to preprocess the trajectories. 

<img width="631" alt="Screenshot 2024-06-27 at 10 23 42" src="https://github.com/PedestrianDynamics/PedPy/assets/5772973/0fc11514-4eb7-450b-8027-b01f9a49689a">

Figure taking from [Smoothing Trajectories of People’s Heads](https://link.springer.com/chapter/10.1007/978-3-030-55973-1_3)

**Comment by schroedtert** [comment]
Great idea! Do you have any particular filters in mind?

**Reply by chraibi** [reply]
I thought at the Savitzky-Golay filter, 
the Central Moving Average (CMA) and the Spline Interpolation of Inflection Points (SIP).

**Comment by MaikFZJ** [comment]
See also discussion https://github.com/PedestrianDynamics/PedPy/discussions/301

Distinguish between
- removing jitter
- detecting steps
- calculate main movement direction by
  - using step detection
  - spline interpolation, inflection point (Bernhard Steffen) (1)
  - Savitzky-Golay filter (Jiayue Wang) (2)
  - moving convex hull (Jana Pick, Janine Klein) (3)
  - moving average

One should take velocity into account when looking for step/stride.

(1)
https://arxiv.org/abs/0911.2165
subfolder in old JPSreport

(2)
Wang J, Weng W, Boltes M, Zhang J, Tordeux A and Ziemer V 2018 Step
styles of pedestrians at different densities J. Stat. Mech. Theory Exp. 2018
023406
Wang J, Boltes M, Seyfried A, Zhang J, Ziemer V and Weng W 2018 Linking
pedestrian flow characteristics with stepping locomotion Phys. A Stat. Mech. its
Appl. 500 106–20

(3)
https://juser.fz-juelich.de/record/888120

Some source about smoothing can be found in
https://juser.fz-juelich.de/record/888120

**Comment by MaikFZJ** [comment]
Have also a look to:
Jia, X.; Feliciani, C.; Yanagisawa, D.; Nishinari, K. Experimental Study on the Evading Behavior of Individual Pedestrians When Confronting with an Obstacle in a Corridor. Physica A: Statistical Mechanics and its Applications 2019, 531, 121735. https://doi.org/10.1016/j.physa.2019.121735.
